### PR TITLE
Use channel in messaging between goroutine instead of zmq.REQ

### PR DESCRIPTION
**Pull request by rickmak**
This will resolve socket exhaust problem on high concurrency.

connects #160

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -48,7 +48,8 @@ before-test:
 test:
 	# Run `go install` to compile packages to speed up test process
 	$(DOCKER_COMPOSE_RUN) go install $(GO_BUILD_ARGS)
-	$(DOCKER_COMPOSE_RUN) go test $(GO_BUILD_ARGS) -cover -timeout $(GO_TEST_TIMEOUT) ./pkg/...
+	GOMAXPROCS=4 $(DOCKER_COMPOSE_RUN) go test $(GO_BUILD_ARGS) -cover -timeout $(GO_TEST_TIMEOUT) ./pkg/...
```
Comment (cheungpat) [inline Makefile]: please do this instead:
```
$(DOCKER_COMPOSE_RUN) go test $(GO_BUILD_ARGS) -cover -timeout $(GO_TEST_TIMEOUT) -cpu 1,4 ./pkg/...
```

**Comment by rickmak** [comment]
Resubmitted at #181

**Comment by cheungpat** [comment]
don’t know what you are doing

**Comment by cheungpat** [comment]
I am going to merge this anyway.

**Pull request merged**